java/serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java

```diff
@@ -52,126 +49,77 @@
   private final OnlineRetrieverV2 retriever;
   private final RegistryRepository registryRepository;
   private final OnlineTransformationService onlineTransformationService;
+  private final String project;
 
   public OnlineServingServiceV2(
       OnlineRetrieverV2 retriever,
       Tracer tracer,
       RegistryRepository registryRepository,
-      OnlineTransformationService onlineTransformationService) {
+      OnlineTransformationService onlineTransformationService,
+      String project) {
     this.retriever = retriever;
     this.tracer = tracer;
     this.registryRepository = registryRepository;
     this.onlineTransformationService = onlineTransformationService;
+    this.project = project;
   }
 
   /** {@inheritDoc} */
   @Override
   public GetFeastServingInfoResponse getFeastServingInfo(
       GetFeastServingInfoRequest getFeastServingInfoRequest) {
-    return GetFeastServingInfoResponse.newBuilder()
-        .setType(FeastServingType.FEAST_SERVING_TYPE_ONLINE)
-        .build();
+    return GetFeastServingInfoResponse.newBuilder().build();
   }
 
   @Override
-  public GetOnlineFeaturesResponse getOnlineFeatures(GetOnlineFeaturesRequestV2 request) {
-    // Autofill default project if project is not specified
-    String projectName = request.getProject();
-    if (projectName.isEmpty()) {
-      projectName = "default";
-    }
-
+  public ServingAPIProto.GetOnlineFeaturesResponseV2 getOnlineFeatures(
+      ServingAPIProto.GetOnlineFeaturesRequest request) {
     // Split all feature references into non-ODFV (e.g. batch and stream) references and ODFV.
-    List<FeatureReferenceV2> allFeatureReferences = request.getFeaturesList();
-    List<FeatureReferenceV2> featureReferences =
+    List<FeatureReferenceV2> allFeatureReferences = getFeaturesList(request);
+    List<FeatureReferenceV2> retrievedFeatureReferences =
         allFeatureReferences.stream()
             .filter(r -> !this.registryRepository.isOnDemandFeatureReference(r))
             .collect(Collectors.toList());
+    int userRequestedFeaturesSize = retrievedFeatureReferences.size();
+
     List<FeatureReferenceV2> onDemandFeatureReferences =
         allFeatureReferences.stream()
             .filter(r -> this.registryRepository.isOnDemandFeatureReference(r))
             .collect(Collectors.toList());
 
     // Get the set of request data feature names and feature inputs from the ODFV references.
-    Pair<Set<String>, List<FeatureReferenceV2>> pair =
-        this.onlineTransformationService.extractRequestDataFeatureNamesAndOnDemandFeatureInputs(
-            onDemandFeatureReferences, projectName);
-    Set<String> requestDataFeatureNames = pair.getLeft();
-    List<FeatureReferenceV2> onDemandFeatureInputs = pair.getRight();
+    List<FeatureReferenceV2> onDemandFeatureInputs =
+        this.onlineTransformationService
+            .extractRequestDataFeatureNamesAndOnDemandFeatureInputs(onDemandFeatureReferences)
```

Comment:
Think with your new context proto, we can clean up a bunch of onlinetransformationservice. e.g. are we still using the "getLeft()" of the extractRequestDataFeatureNamesAndOnDemandFeatureInputs?

Maybe add TODOs to clean up? I also don't really see a test we have for on demand.

Reply:
We can add the on demand tests after this PR merges

Reply:
Yeah, there were no tests. That bothers me as well, but this PR is already big enough.
ToDo added.